File: src/main/java/io/xol/chunkstories/api/exceptions/content/mods/NotAllModsLoadedException.java
//
// This file is a part of the Chunk Stories API codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package io.xol.chunkstories.api.exceptions.content.mods;

import java.util.Collection;

public class NotAllModsLoadedException extends Exception
{
	Collection<ModLoadFailureException> failed;
	
	public NotAllModsLoadedException(Collection<ModLoadFailureException> failed)
	{
		this.failed = failed;
	}
	
	public String getMessage()
	{
		String message = "Some mods failed to load : \n";
		
		for(ModLoadFailureException e : failed)
			message += e.getMessage() + "\n";
		
		return message;
	}
	
	/**
	 * 
	 */
	private static final long serialVersionUID = 5136184783162902334L;

}

File: src/main/java/xyz/chunkstories/api/exceptions/content/mods/NotAllModsLoadedException.kt
//
// This file is a part of the Chunk Stories API codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.api.exceptions.content.mods

class NotAllModsLoadedException(internal var failed: Collection<ModLoadFailureException>) : Exception() {

    override val message: String?
        get() {
            var message = "Some mods failed to load : \n"

            for (e in failed)
                message += e.message + "\n"

            return message
        }

    companion object {

        /**
         *
         */
        private val serialVersionUID = 5136184783162902334L
    }

}
